src/applications/ElementalChat/store/elementalChat.js

```diff
@@ -129,14 +129,32 @@ export default {
       log("Updating Handle");
       rootState.needsHandle = true;
     },
-    getStats: async ({ rootState, dispatch, commit }) => {
+    getStats: async ({ state, rootState, dispatch, commit }) => {
       log("Getting Stats...");
       commit("loadStats");
       callZome(dispatch, rootState, "chat", "agent_stats", null, 60000)
-        .then(stats => {
+        .then(async stats => {
           log("stats zomeCall done");
-          console.log("Peer stats", stats);
-          commit("setStats", stats);
+          log("Peer stats: ", stats);
+          let channels = 0;
+          let messages = 0;
+          await rootState.hcDb.elementalChat.get("General").then(async c => {
+            if (c === undefined) channels = 0;
+            else channels = c.length;
+            for (let i = 0; i < c.length; i++) {
+              await rootState.hcDb.elementalChat
+                .get(c[i].channel.uuid)
+                .then(m => {
+                  if (m === undefined) messages += 0;
```

Comment:
This seems a little odd, why not: `if ((m !== undefined) && (m.messages !== undefined)) mesages += m.messages.length;`

Reply:
sure same thing in a different direction. updating now